src/components/utils/ErrorHandler.js

```diff
@@ -0,0 +1,235 @@
+
+function ContactSupport(props) {
+    const { baseId } = props;
+    return (
+        <Button
+            id={build(baseId, ids.CONTACT_SUPPORT_BUTTON)}
+            color="primary"
+            startIcon={<LiveHelpIcon />}
+            onClick={() => window.Intercom("show")}
+        >
+            {getMessage("contactSupport")}
+        </Button>
+    );
+}
+
+function ErrorHandler(props) {
+    const { errorObject, baseId } = props;
+    const router = useRouter();
+    const classes = useStyles();
+    const errBaseId = build(baseId, ids.ERROR_HANDLER);
+    if (!errorObject || !errorObject.response) {
+        return (
+            <Card id={errBaseId}>
+                <CardHeader
+                    title={
+                        <Typography color="error" variant="h6">
+                            <ErrorIcon /> {getMessage("oops")}
+                        </Typography>
+                    }
+                    subheader={
+                        <Typography color="error">
+                            {getMessage("unexpectedError")}
+                        </Typography>
+                    }
+                />
+                <Divider oritentation="horizontal" />
+                <CardContent className={classes.cardContent}>
+                    <Grid container spacing={2} className={classes.container}>
+                        <ClientInfo baseId={baseId} />
+                    </Grid>
+                </CardContent>
+                <Divider oritentation="horizontal" />
+                <CardActions>
+                    <ContactSupport baseId={baseId} />
+                </CardActions>
+            </Card>
+        );
+    }
+    const errorResponse = errorObject.response;
+    if (errorResponse?.status === 401) {
+        return (
+            <Card>
+                <CardHeader
+                    title={
+                        <>
+                            <Typography component="span" variant="h5">
+                                <ErrorIcon color="primary" />
+                                {getMessage("signInReqd")}
+                            </Typography>
+                        </>
+                    }
+                />
+                <Divider orientation="horizontal" />
+                <CardContent className={classes.cardContent}>
+                    <Typography variant="h6" className={classes.signIn}>
+                        {getMessage("youMust")}
+                        <Link
+                            id={build(errBaseId, ids.SIGN_IN_LINK)}
+                            color="primary"
+                            onClick={() => {
+                                router.push(
+                                    `/${NavigationConstants.LOGIN}${router.asPath}`
+                                );
+                            }}
+                        >
+                            {getMessage("signIn")}
+                        </Link>
+                        {getMessage("viewThisPage")}
```

Comment:
This isn't really i18n friendly, but I don't think you can pass a `Link` object to `getMessage` 😕 

As an alternative, we could have a single "You must sign in to view this page" message, and wrap the entire thing in the `Link` 🤔

Or maybe use a primary color `Button` with this message as the label?